### DISCUSSION TITLE
가격과 호텔 날짜로 인한 고민거리

### DISCUSSION THREAD

**Discussion by Aleexender** [SCBJ-7/SCBJ-BE#8] [Ideas]
## - 고민사항
구매내역같은걸 확인할때 어떻게 하면 좋을까요?

1.Product 혹은 Reservation 에서 해당 날짜와 가격을 저장을 해서 데이터가 점점 쌓이면 스케쥴링을 통해서 지우기 
2. 따로 Price라는 테이블로 상품에대한 가격과 날짜를 저장해서 관리하고 쌓이면 스케쥴링을 통해서 지우기 

혹은 
MongoDB를 이용해서 따로 저장하는 것들을 관리하기

#### 위에 있는 사항들은 둘다 **DELETE**를 사용해서 관리해야합니다.
**DELETE**를 해야하는 이유?
- MySQL에 대한 데이터 정리 및 물리적인 공간 확보

#### 1 의 장점
- 굳이 Join을 통해서 가져올 필요 없이 그냥 조회만 해서 가져오면 된다.

#### 1의 단점
- 컬럼 수정시에 LOCK 이 동작할수 있음

===============================================================

#### 2의 장점
- 테이블 수정시 편리함

#### 2의 단점
- Join을 이용한 조회이기 때문에 1번 보다 속도가 느리다 

MongoDB 를 이용할시 위 두개의 고민점이 사라질수 있긴하나 시도해보지 않아 시간비용이 발생할거 같습니다.

**Comment by wocjf0513** [comment]
따로 NoSQL에 저장하는 게 맞을 거 같긴 한데, 충분히 공수가 많이 들 거 같아서 어렵네요.

**Reply by Aleexender** [reply]
추후에 NoSQL에 저장해도 나쁘지 않을꺼 같긴해용!

**Comment by Kwonminwoo** [comment]
delete에 대해 생각해보지 못했는데 좋은 안건인 것 같네요
